src/odemis/driver/tlaptmf.py

```diff
@@ -303,7 +303,10 @@ def SendMessage(self, msg, dest=0x50, src=1, p1=None, p2=None, data=None):
             while True:
                 trials += 1
                 try:
-                    logging.debug("Sending: '%s'", ", ".join("%02X" % ord(c) for c in com))
+                    # Iterate through com.decode('latin1', 'backslashreplace') to allow compatibility with python2 and python3.
+                    # Iteration through a byte str in python3 automatically returns the code point,
+                    # while python2 requires explicit conversion with ord().
+                    logging.debug("Sending: '%s'", ", ".join("%02X" % ord(c) for c in com.decode('latin1', 'backslashreplace')))
```

Comment:
With "weird" characters, it'll go wrong, as it'll show each value of the escaped string. I guess you could just decode to latin1 without special handler, or, prettier, convert to a `bytearray`, like:
`"%02X" % ord(c) for c in bytearray(com)`

You could also make a dedicated function, which behaves differently depending on the python version. Something like `_msg_to_str()`. You'd write 2 separate functions `_msg_to_str_py2()` and `_msg_to_str_py3()`, and assign one of them at init, based on the Python version.

Reply:
bytearray requires a second argument for the encoding in python3, so it's quite similar to .decode('latin1'). I think I'm going to stick with this and remove the backslashreplace handler

Reply:
actually, I think bytearray wouldn't even work in python3 because com is already bytes.

Reply:
Sorry, actually I meant. `"%02X" % c for c in bytearray(com)`.
It does work on Python 2 and Python 3 (I tried!). com is a string on python2 and bytes on python3. In both cases, this creates the right bytearray, which is immediately iterated as ints.
Please update all the logging which use `ord()` .